packages/x-tree-view/src/internals/plugins/useTreeViewItems/useTreeViewItems.utils.ts

```diff
@@ -50,3 +54,155 @@ export const isItemDisabled = (
 
   return false;
 };
+
+type State = UseTreeViewItemsState<any>['items'];
+export function buildItemsState(parameters: BuildItemsStateParameters): State {
```

Comment:
I splitted this function into two:

- one that only generates the lookups to set the list of children of an item
- one that generates the whole item state (and uses the 1st one to do so)

It allows us to not generate a full state when we want to insert children in the tree